fhirstore/fhirstore.py

```diff
@@ -260,8 +260,16 @@ def validate(self, resource):
             raise Exception(f"missing schema for resource {resource}")
         validate(instance=resource, schema=schema)
 
-    def search(self, resource_type, params, offset=0, result_size=100, elements=None):
-
+    def search(
```

Comment:
coming back on our conversation yesterday: you could unite all these parameters in a single `Search` class and pass an instance of this class as parameter of this function

Reply:
you could for instance have a method `Search.build_es_query` that transform these parameters into an elasticsearch query object

Reply:
I will do an issue about this which I'll address in another PR.